server/data/data-cassandra/src/main/java/org/apache/james/user/cassandra/CassandraUsersRepository.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.user.cassandra;
+
+import javax.inject.Inject;
+
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.configuration2.tree.ImmutableNode;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.user.lib.UsersRepositoryImpl;
+
+
+public class CassandraUsersRepository extends UsersRepositoryImpl<CassandraUsersDAO> {
+    @Inject
+    public CassandraUsersRepository(DomainList domainList, CassandraUsersDAO usersDAO) {
+        super(domainList, usersDAO);
+    }
+
+    @Override
+    public void configure(HierarchicalConfiguration<ImmutableNode> config) throws ConfigurationException {
+        super.configure(config);
+        usersDAO.configure(config);
```

Comment:
👎 

1. We tend to associate a pojo to all new confs being written and inject that pojo. It makes things easier (conf details centralisation, beautiful builders for the conf to discover options from within an ide, easier injects, static congiguration)

2. Inheritence should imo be reserved for last resort needs. This do not sound like the case here.

Why don t we write a pojo summurizing this conf and pass it to CassandraUserDao constructor?

Then the guice module would be free to parse it from usersrepository.xml file if it wants, without inheritence.

Reply:
I don't quite understand. You are saying `HierarchicalConfiguration<ImmutableNode> config` is bad and should be a POJO instead? And there should not be a `CassandraUsersRepository` class? Could you point me to some other class where its done right please, just for reference?

Reply:
Never mind, I see what you mean, e.g. like its done for `LdapRepositoryConfiguration`. However, the generic `UsersRepositoryImpl` still expects `configure(HierarchicalConfiguration<ImmutableNode> configuration)` for `virtualHosting`, `administratorId`, `verifyFailureDelay`, all of which are private members. Should these be protected instead? Should `UsersRepositoryImpl` accept a second type parameter for the configuration type? Then there would have to be something like `CassandraRepositoryConfiguration extends CommonRepositoryConfigration`, and the same for all other repository types. All doable, but this starts to exceed the scope of this feature PR. I feel that rewriting it all to configuration pojos should be a separate ticket. Please advise how to proceed.

Reply:
IMO `UsersRepositoryImpl` and CassandraUserDAO should be configured separatly. The fact that they are backed by the same file is a detail.

Regarding the POJO topic, this is IMO a nice to have. I would not promote it to a MUST have.

Reply:
Fixed in latest push. `CassandraUsersRepositoryModule` now provides `CassandraRepositoryConfiguration` which gets injected into `CassandraUsersDAO`.
The added class `CassandraUsersRepository` is gone, everybody uses `UsersRepositoryImpl<CassandraUsersDAO>` again.